tests/on_wlm/test_colocated_model.py

```diff
@@ -31,95 +31,123 @@
 
 from smartsim import Experiment, status
 
+launcher = pytest.test_launcher
+supported_dbs = ["uds", "tcp", "deprecated"]
+
 # retrieved from pytest fixtures
-if pytest.test_launcher not in pytest.wlm_options:
+if launcher not in pytest.wlm_options:
     pytestmark = pytest.mark.skip(reason="Not testing WLM integrations")
 
+def _setup_test_colo(fileutils, db_type, exp_name, db_args, launcher):
```

Comment:
Maybe the code would be simpler if the experiment was created outside of this function and the experiment was passed in by reference?  Then no need to return tuple, etc.

Reply:
I'm inclined to keep this as is since we do need both and otherwise we have two very small functions. Happy to revisit though

Reply:
Either way is fine.  What I was thinking wouldn't involve two functions -- I was thinking that the Experiment should be created as part of the test (see snippet below).  ``launcher`` and ``exp_name`` are passed in only to call the experiment constructor.  It might make it closer to the ideal of a function only doing one task so the code is more reusable later if this function wasn't also creating an experiment.

```python
@pytest.mark.parametrize("db_type", supported_dbs)
def test_colocated_model_pinning_auto_1cpu(fileutils, db_type):
     exp = Experiment(name, launcher)
     _setup_colo_model(fileutils, db_args, db_args, exp)
```

Reply:
Ohhh I see what you mean. That makes sense.